lib/liquid/tags/render.rb

```diff
@@ -35,15 +35,10 @@ def render_tag(context, output)
       template_name = context.evaluate(@template_name_expr)
       raise ArgumentError, options[:locale].t("errors.argument.include") unless template_name
 
-      partial = PartialCache.load(
-        template_name,
-        context: context,
-        parse_context: parse_context
-      )
-
       context_variable_name = @alias_name || template_name.split('/').last
 
       render_partial_func = ->(var) {
+        partial = load_partial(template_name, context)
```

Comment:
I don't think this needs to go inside `render_partial_func`, when using the `for` syntax it will run this for every iteration.